BUILDING.md

```diff
@@ -0,0 +1,39 @@
+# Building Protocol Buffers
+
+These are instructions for building the open source release of protobuf from source.
+
+Given the right tools, this project can be built on Linux, Mac OS X, or Windows though Linux is by far the most commonly used and best tested platform.
+
+## Prerequisites
+
+Likely incomplete
+
+* git
+* JDK 8 or JDK 11. JDK 17 probably doesn't work.
+* bazel
+* clang
+
+At the top level, this project is built with bazel. Java parts 
+are built with Maven. All languages assume that protoc is built and
```

Comment:
This is about to no longer be true. We build maven artifacts using bazel via rules_jvm_external.

Reply:
Let's update when that's no longer true. Or is https://github.com/protocolbuffers/protobuf/blob/master/java/BUILD already incorrect?

Reply:
That file is the bazel releases. But I think the bigger question here is who is this documentation for? Is this for users building the library from scratch? They can use bazel already. But if you want to mention Maven here I think you should also provide instructions on how to build with maven.

Reply:
That is the _corrected_ BUILD file that makes use of rules_jvm_external.

Reply:
This is for developers building this repo from scratch. We need complete instructions for checkout and build. Right now we don't have that. I'll remove the part about Maven unless I discover we're still using it.